azure/datalake/store/lib.py

```diff
@@ -136,60 +131,63 @@ def auth(tenant_id=None, username=None,
 
     if not client_secret:
         client_secret = os.environ.get('azure_client_secret', None)
+        contextPub = msal.PublicClientApplication(client_id=client_id, authority=authority+tenant_id)
 
-    # You can explicitly authenticate with 2fa, or pass in nothing to the auth call
-    # and the user will be prompted to login interactively through a browser.
+    if tenant_id is None or client_id is None:
+        raise ValueError("tenant_id and client_id must be supplied for authentication")
 
-    @retry_decorator_for_auth(retry_policy=retry_policy)
+    contextClient = msal.ConfidentialClientApplication(client_id=client_id, authority=authority+tenant_id, client_credential=client_secret)
```

Comment:
```suggestion
    else:
        contextClient = msal.ConfidentialClientApplication(client_id=client_id, authority=authority+tenant_id, client_credential=client_secret, http_cache=_http_cache)
```
Similar suggestion to the comment I put for `PublicClientApplication`. Also, this should be in an `else` clause, so that you can save overhead for constructing 2 MSAL classes but using only one.